luaui/Widgets/gui_gridmenu.lua

```diff
@@ -2168,22 +2340,20 @@ local function handleButtonHover()
 		if hovering then
 			-- cells
 			if hoveredCellID then
-				local uDefID = cellCmds[hoveredCellID].id * -1
-				local cellIsSelected = (
-					activeCmd
-						and cellCmds[hoveredCellID]
-						and activeCmd == cellCmds[hoveredCellID].name
-				)
+				local hoveredCell = cellRects[hoveredCellID]
+				local uDefID = hoveredCell.opts.uDefID
+				local cellIsSelected = activeCmd == -uDefID
+				local queuenr = hoveredCell.opts.queuenr
```

Comment:
all the draw highlight stuff could probably be broken out into another function.

Reply:
Can you clarify?

Reply:
in an attempt to keep things more organized, the function could be cut in half between logic and drawing

Reply:
All this garbage has been removed, I might revisit the showy click effects later.